platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ImportMigrationConfigurationAdminContext.java

```diff
@@ -0,0 +1,312 @@
+    @SuppressWarnings("PMD.DefaultPackage")
+    Configuration createConfiguration(ImportMigrationConfigurationAdminEntry entry)
+            throws IOException {
+        // Question: should we use the bundle location that was exported???
+        if (entry.isManagedServiceFactory()) {
+            return configurationAdmin.createFactoryConfiguration(entry.getFactoryPid(), null);
+        }
+        return configurationAdmin.getConfiguration(entry.getPid());
+    }
+
+    // PMD.DefaultPackage - designed to be called only from ImportMigrationConfigurationAdminEntry within this package
+    @SuppressWarnings("PMD.DefaultPackage")
+    Configuration getMemoryConfiguration(ImportMigrationConfigurationAdminEntry entry) {
+        final String fpid = entry.getFactoryPid();
+
+        if (fpid != null) {
+            final List<Configuration> mcfgs = memoryFactoryServices.get(fpid);
+
+            if (mcfgs == null) {
+                return null;
+            }
+            // search for it based on the felix file install property
+            final Path epath = getPathFromConfiguration(entry.getProperties(),
+                    () -> String.format("path '%s'", entry.getPath()));
+
+            if (epath != null) {
+                for (final Iterator<Configuration> i = mcfgs.iterator(); i.hasNext(); ) {
+                    final Configuration mcfg = i.next();
+                    final Path mpath = getPathFromConfiguration(mcfg.getProperties(),
+                            () -> String.format("configuration '%s'", mcfg.getPid()));
+
+                    if (epath.equals(mpath)) {
+                        // remove it from memory list and clean the map if it was the last one
+                        i.remove();
+                        if (mcfgs.isEmpty()) {
+                            memoryFactoryServices.remove(fpid);
+                        }
+                        return mcfg;
+                    }
+                }
+            }  // else - this means we will not be able to correlate an exported managed service factory
+            //           with its counterpart here, as such we will be forced to treat it as a new one
+            return null;
+        }
+        return memoryServices.remove(entry.getPid());
+    }
+
+    private ImportMigrationConfigurationAdminEntry proxy(ImportMigrationEntry entry) {
+        final Path path = entry.getPath();
+        final String extn = FilenameUtils.getExtension(path.toString());
+        final PersistenceStrategy ps = admin.getPersister(extn);
+
+        if (ps == null) {
+            getReport().record(new MigrationException(
+                    "Import error: persistency strategy [%s] for configuration [%s] is not defined.",
+                    extn,
+                    path));
+        } else {
+            final Dictionary<String, Object> properties;
+            InputStream is = null;
+
+            try {
+                is = entry.getInputStream()
```

Comment:
✏️ This block - from line 210-216 - can be simplified to
```
                is = entry.getInputStream()
                        .orElseThrow(() -> new MigrationException(
                                "Import error: failed to read configuration [%s]; not exported.",
                                path));
```

Reply:
How did I miss that one 😄